spidermonkey_91: drop

pyrox0 opened the pull request:
Fixed up version of #437376

It has been EOL for several years now, and the last dependant(couchdb3) no longer requires it as of #436932.

I've also removed all conditionals in the common.nix build file that were for versions older than 115, since we no longer ship any versions older than that. I would like to remove 115 and 128 soon as well, since as of commit, they reach end of life in 3 weeks(in fact, 115 is EOL for all platforms we support afaik). Also, I removed the `meta.broken` attribute, since that only talks about versions lower than 115, our current minimum. If it turns out some packages are still broken on darwin, I would be glad to add it back for the versions that need it.


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [x] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

pyrox0 committed:
spidermonkey_91: drop

It has been EOL for several years now, and the last dependant(couchdb3)
no longer requires it as of #436932.

I've also removed all conditionals in the common.nix build file that
were for versions older than 115, since we no longer ship any versions
older than that. I would like to remove 115 and 128 soon as well, since
as of commit, they reach end of life in 3 weeks(in fact, 115 is EOL for
all platforms we support afaik)

pyrox0 commented:
I believe I have fixed the darwin build issue reported at https://github.com/NixOS/nixpkgs/pull/437376#issuecomment-3226698427

Scrumplex commented:
`pkgs/servers/http/couchdb/3.nix` uses spidermonkey_91

Scrumplex commented on pkgs/top-level/aliases.nix:
```diff
@@ -2173,6 +2173,7 @@ mapAliases {
   # spidermonkey is not ABI upwards-compatible, so only allow this for nix-shell
   spidermonkey = throw "'spidermonkey' has been renamed to/replaced by 'spidermonkey_91'"; # Converted to throw 2024-10-17
   spidermonkey_78 = throw "'spidermonkey_78' has been removed because it was unused."; # Added 2025-02-02
+  spidermonkey_91 = throw "'spidermoney_91 has been EOL since 2022/09"; # Added 2025-08-26
```
```suggestion
  spidermonkey_91 = throw "'spidermoney_91' is EOL since 2022/09"; # Added 2025-08-26
```

pyrox0 commented:
> Restarted the tests since [4acde72](https://github.com/NixOS/nixpkgs/commit/4acde7282a63863a21f297da0f0db4c643deac58) is now in staging. The new failure:
> <img alt="图片" width="1952" height="262" src="https://private-user-images.githubusercontent.com/20080233/482586452-6b253333-3b34-4a72-8312-ededfe0eb5ab.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYyOTY4MDAsIm5iZiI6MTc1NjI5NjUwMCwicGF0aCI6Ii8yMDA4MDIzMy80ODI1ODY0NTItNmIyNTMzMzMtM2IzNC00YTcyLTgzMTItZWRlZGZlMGViNWFiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODI3VDEyMDgyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThkZDQzN2Y2ODEwOTdjNmE0NGYzNTlkNTI0NTAyMzA5MTJhY2I4ZjE2MmRhMmRjMDM2ZGM3M2YyNGRkMzRhN2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3qUjf0x4yV_fSP7XMp-kRX-gQvO3sqLGFxFACA9h_dE">
> 
> (As one of the maintainers of this package, I won't mind keeping `meta.broken` untouched if nobody actually needs this on darwin 🤷)

yeah, that was a typo on my end, whoops. Fixed.

pyrox0 commented:
okay, darwin will be without spidermonkey again, I don't have the energy or hardware to debug it.

mweinelt commented:
For 140 you can rebase `pkgs/build-support/build-mozilla-mach/142-relax-apple-sdk.patch`.

pyrox0 committed:
spidermonkey_140: relax SDK requirements for darwin

Upstream wants SDK 15.5, we only have 15.2, but that works fine.

bobby285271 committed:
spidermonkey_140: make apple-sdk patch unconditional

To prevent rotten patches.

bobby285271 committed:
spidermonkey_115: mark as broken on darwin again

This still does not build.

> DEBUG: Executing: `/nix/store/k4mvgh65zn6xb4f6zs9kypz4lq8sgmk3-clang-wrapper-19.1.7/bin/clang -isysroot /nix/store/qvbwadc5f7akpzaf3psd41wjl5jq8333-apple-sdk-14.4/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.sdk -mmacosx-version-min=11.3 -std=gnu99 -fuse-ld=lld -Wl,--version`
> ERROR: Failed to find an adequate linker

bobby285271 commented:
Thanks!
